src/consensus/block_producer.rs

```diff
@@ -430,15 +428,31 @@ where
             }
         }
         if slice_capacity_left < MAX_TRANSACTION_SIZE {
-            break duration_left.saturating_sub(start_time.elapsed());
+            break;
         }
-    };
+    }
 
-    // TODO: not accounting for this potentially expensive operation in duration_left calculation above.
+    let num_txs = txs.len();
     let txs = bincode::serde::encode_to_vec(&txs, bincode::config::standard())
         .expect("serialization should not panic");
     let payload = SlicePayload::new(parent, txs);
-    (payload, ret)
+
+    // idle if necessary
+    // TODO: instead actually spend more time waiting for and selecting transactions
+    if start_time.elapsed() < min_duration && !preempt_receiver.is_terminated() {
```

Comment:
This would be true if the slice is already full?  In that case, if we receive more txs, where will be put them?  

If a potential idea is to buffer up the additional txs in an internal queue, we will have to be careful not to buffer up more than we will be able to fit in the remaining slices of this slot and potentially remaining slots of the leader window.